Docs: API overview text clarity & kv2 mention... #16746

aphorise opened the pull request:
Corrected text and terminology. Relates to #6378 & should allow for closure of that issue too.
<img width="1158" alt="Screenshot 2022-08-16 at 19 23 20" src="https://user-images.githubusercontent.com/974854/184941452-2b2c680a-b6d5-4db6-85aa-e5dc672499f6.png">

benashz commented on website/content/api-docs/index.mdx:
```diff
@@ -123,9 +126,9 @@ endpoints that consume parameters use `POST` instead and put the parameters in
-parameter `list=true`, or you can use the `LIST` HTTP verb. For the `kv`
-backend, listing is allowed on directories only, and returns the keys in the
-given directory:
+parameter `list=true`, or you use the `LIST` HTTP verb. For the `kv` secrets 
```
You may want to put caveat portion of this statement in italics.

aphorise replied:
@benashz do you mean the `or` or which portion do you think should be made italic?

benashz replied:
I think we can ignore this comment. I may have misinterpreted it.

benashz commented on website/content/api-docs/index.mdx:
```diff
@@ -192,12 +195,15 @@ addition, the Vault SDK always adds this header to every request.
 ## Help
 
-To retrieve the help for any API within Vault, including mounted backends, auth
+To retrieve the help for any API within Vault, including mounted engine, auth
```
```suggestion
To retrieve the help for any API within Vault, including mounted engines, auth
```

benashz commented on website/content/api-docs/index.mdx:
```diff
@@ -192,12 +195,15 @@ addition, the Vault SDK always adds this header to every request.
+To retrieve the help for any API within Vault, including mounted engine, auth
 methods, etc. then append `?help=1` to any URL. If you have valid permission to
-access the path, then the help text will be return a markdown-formatted block in the `help` attribute of the response.
+access the path, then the help text will be return a markdown-formatted block 
```
```suggestion
access the path, then the help text will be returned as a markdown-formatted block 
```

aphorise commented on website/content/api-docs/index.mdx:
```diff
@@ -90,14 +86,21 @@ $ curl \
-detailed information on its API; this simply provides a general overview.
+There are documented difference in the path schemes as well as other 
+functionality which may be applicable as for example with [`KVv2` compared to `KVv1`].
 
```
There was some feedback originally on how difference in KV1 / KV2 is not explicit spelt out. The original does seem a bit generic - but please do suggest any alterative that you may see as more suitable.

benashz replied:
You could provide a table of differences, or comparable usage examples for each.

aphorise replied:
@benashz I think as an intro the way it is at present can suffice and just listing differences KVv2 in any table could be the bulk of the page (especially since KV2 has so much more) rather than this page just being a quick overview and intro into how API calls are made.

aphorise commented on website/content/api-docs/index.mdx:
```diff
@@ -90,14 +86,21 @@ $ curl \
+
+The demonstration below are using the `KVv1` secrets engine, which is a 
+simple Key/Value store. Please read the API documentation of secret engine or 
+authentication engine for detailed information on their use. 
```
I will correct this on the next commit.

benashz commented on website/content/api-docs/index.mdx:
```diff
@@ -90,14 +86,21 @@ $ curl \
-Vault is via JSON. Vault will set the `Content-Type` header appropriately but
-does not require that clients set it.
+All request data, body and response data to and from Vault is JSON except in a 
+few exceptional cases that are documented. Vault sets the `Content-Type` header 
```
I would omit the last bit about being `documented` unless  you can provide a link to said documentation.

aphorise replied:
Ok let me see what I can do here - the original and current docs do note the same exception but you are right without linking it's not helpful - all those other cases are by user request in any case (I was thinking like some PKI endpoints that provide binary).

benashz commented on website/content/api-docs/index.mdx:
```diff
@@ -90,14 +86,21 @@ $ curl \
+
+Authentication and Secret engines, that are also referred to as backends or 
+mounts, differ in their implementation of APIs according to functionality and 
+how they're authored.
```
This sentence is hard to parse. I am not sure what it is  trying to be  conveyed by this statement. Are you trying to say that each API may have a unique implementation/design?

benashz commented on website/content/api-docs/index.mdx:
```diff
@@ -21,12 +21,6 @@ is, `sys/` routes) change very infrequently, but various secrets engines/auth
 methods/etc. sometimes have minor changes to accommodate new features as
 they're developed.
 
```
The request PATH difference seems to be one of the most obvious differences between v1 and v2, but this change seem to be dropping this bit of information all together?

aphorise replied:
@benashz I've added a specific note on URI path differences and features.

aphorise commented on website/content/api-docs/index.mdx:
```diff
@@ -90,14 +86,21 @@ $ curl \
+how they're authored.
+
+The demonstration below are using the `KVv1` secrets engine, which is a 
+simple Key/Value store. Please read the API documentation of secret engine or 
```
added KVv1 API links

aphorise commented:
Hey @benashz can you kindly recheck again.

aphorise commented:
Hey @benashz further to your feedback I've added the KV2 example explicitly.

I've also adjusted the grammer a bit as there are:
 - `query` should be `query string` 
 - over use of the word `via` in various places to mean `in` or `using` or `with`, etc.
 - over use of the word `to` - especially in some places eg:
     >Vault tries **to** (1) adhere **to** (2) these whenever possible, but in some cases may not -- feel free **to** (3) 
file a bug in that case **to** (4) point our attention **to** (5) it!

@yhyakuna I'd appreciate your eyes on this too (and anyone else).

yhyakuna reviewed (approved):
Thank you for catching those.
